### PR TITLE
Making dashboard usable

### PR DESCRIPTION
**Pull request by Christen97**
Give the users a better overview of their current tasks.

### CODE REVIEW

**Comment by mcamp** [comment]
It looks good, can you post a picture of the dashboard to see how it looks like?

**Comment by Christen97** [comment]
> It looks good, can you post a picture of the dashboard to see how it looks like?

The styling isn't totally done yet, and it is still missing some functionalities.

**Comment by Christen97** [comment]
Current state of the dashboard:
DashboardFirstIteration

**Comment by Christen97** [comment]
Dashboard with no data:
DashboardEmpty 2022-09-29 211010
Dashboard with Web Technologies course:
DashboardWithProd
Dashboard full of data:
DashboardMaxData

File: app/Http/Controllers/DashboardController.php [hunk 1]
```diff
@@ -2,13 +2,40 @@
...
-    public function index() : View
+    public function index(): View
     {
-        return view('dashboard');
+        $courses = auth()->user()->courses()->get();
+        $tasks = Task::whereIn('course_id', $courses->pluck('id'))->where('ends_at', '>=', now())->assignments()->orderBy('ends_at', 'asc')->get();
+        $nextAssignment = $tasks->first();
+        $courseAssignments = Task::assignments()->whereIn('course_id', $courses->pluck('id'))->get();
+        $exercises = Task::exercises()->whereIn('course_id', $courses->pluck('id'))->orderBy('starts_at', 'asc')->limit(5)->get();
+        $completedAssignments = Grade::where('value', '=', 'passed')->get();
```
Comment (jazerix) [inline app/Http/Controllers/DashboardController.php]: There is no reason to pull all grades here. Only fetch relevant ones.

File: app/Http/Controllers/DashboardController.php [hunk 2]
```diff
@@ -2,13 +2,40 @@
...
 {
-    public function index() : View
+    public function index(): View
     {
-        return view('dashboard');
+        $courses = auth()->user()->courses()->get();
+        $tasks = Task::whereIn('course_id', $courses->pluck('id'))->where('ends_at', '>=', now())->assignments()->orderBy('ends_at', 'asc')->get();
+        $nextAssignment = $tasks->first();
+        $courseAssignments = Task::assignments()->whereIn('course_id', $courses->pluck('id'))->get();
+        $exercises = Task::exercises()->whereIn('course_id', $courses->pluck('id'))->orderBy('starts_at', 'asc')->limit(5)->get();
```
Comment (jazerix) [inline app/Http/Controllers/DashboardController.php]: Use take instead of limit

File: resources/views/dashboard.blade.php [hunk 3]
```diff
@@ -1,7 +1,179 @@
 @extends('master')
 
 @section('content')
-    <div class="flex w-full justify-center py-6">
-        <h2 class="text-xl font-bold text-lime-green-500 dark:text-lime-green-400">This page is still under construction.</h2>
+    <div class="container flex gap-x-20 px-6 mx-auto pt-10">
+        <div class="flex-auto w-1/2">
+            @if(sizeof($courses) == 0)
```
Comment (jazerix) [inline resources/views/dashboard.blade.php]: $courses->count()

File: resources/views/dashboard.blade.php [hunk 4]
```diff
@@ -1,7 +1,179 @@
...
+                                            {{ $course->updated_at->diffForHumans() }}
+                                        </p>
+                                    </a>
+                                </li>
+                            </ol>
+                        @endforeach
+                    </div>
+                </div>
+            @endif
+            @if(sizeof($exercises) == 0)
```
Comment (jazerix) [inline resources/views/dashboard.blade.php]: Same here

File: resources/views/dashboard.blade.php [hunk 5]
```diff
@@ -1,7 +1,179 @@
...
+                    <div class="px-6 pt-5">
+                        <h1 class="text-black dark:text-white text-lg font-semibold">
+                            Exercises
+                        </h1>
+                        <p class="text-black dark:text-gray-500 text-medium">
+                            You currently have no exercises active
+                        </p>
+                    </div>
+                </div>
+            @elseif(sizeof($exercises) < 4)
```
Comment (jazerix) [inline resources/views/dashboard.blade.php]: Same here

File: resources/views/dashboard.blade.php [hunk 6]
```diff
@@ -1,7 +1,179 @@
...
+                                        {{ strip_tags(Str::words($exercise->description, 20)) }}</p>
+                                </div>
+                            </li>
+                        </ol>
+                    @endforeach
+                </div>
+            @endif
+        </div>
+        <div class="flex-auto w-1/2">
+            @if(sizeof($tasks) == 0)
```
Comment (jazerix) [inline resources/views/dashboard.blade.php]: Same here

File: resources/views/dashboard.blade.php [hunk 7]
```diff
@@ -1,7 +1,179 @@
...
+                    <div class="px-6 pt-5">
+                        <h1 class="text-black dark:text-white text-lg font-semibold">
+                            Active assignments
+                        </h1>
+                        <p class="text-black dark:text-gray-500 text-medium">
+                            You currently have no assignments active
+                        </p>
+                    </div>
+                </div>
+            @elseif(sizeof($tasks) < 5)
```
Comment (jazerix) [inline resources/views/dashboard.blade.php]: Same here

**Review by jazerix — CHANGES_REQUESTED** [review]
Looking good almost there ! :)

File: app/Http/Controllers/DashboardController.php (hunk 1 above)
Comment (Christen97) [reply app/Http/Controllers/DashboardController.php]: That is actually from when we had the progress bars, I will just delete it.

**Review by jazerix — APPROVED** [review]

**Pull request merged**